File: backend/src/api/urls/compiler.py
from django.urls import path
from src.api.views.compiler import (
    run_code, submit_solution, compile_code_view,
    task_list, task_detail, task_solved_check, analyze_code, leaderboard,
)

urlpatterns = [
    # Task ro'yxati va detail
    path('tasks/', task_list, name='task-list'),
    path('tasks/<int:task_id>/', task_detail, name='task-detail'),
    path('tasks/<int:task_id>/solved/', task_solved_check, name='task-solved-check'),

    # Submit va run
    path('submit/<int:task_id>/', submit_solution, name='submit-solution'),
    path('submit/', submit_solution, name='submit-solution-body'),
    path('run_code/', run_code, name='run-code'),
    path('compile_code/', compile_code_view, name='compile-code'),

    # AI tahlil
    path('analyze/', analyze_code, name='analyze-code'),

    # Leaderboard
    path('leaderboard/', leaderboard, name='leaderboard'),
]

File: frontend/src/services/compiler.js
import api from './api';

export const getTasks = (params = {}) =>
    api.get('/compiler/tasks/', { params });

export const getTask = (id) =>
    api.get(`/compiler/tasks/${id}/`);

export const submitCode = (taskId, code) =>
    api.post(`/compiler/submit/${taskId}/`, { code });

export const runCode = (userCode, testCode) =>
    api.post('/compiler/run_code/', { user_code: userCode, test_code: testCode });

export const analyzeCode = (taskId, userCode, actualOutput) =>
    api.post('/compiler/analyze/', {
        task_id: taskId,
        user_code: userCode,
        actual_output: actualOutput,
    });
